diff --git a/ports/openssl/unix/portfile.cmake b/ports/openssl/unix/portfile.cmake
index f959c6a00..6c2049c88 100644
--- a/ports/openssl/unix/portfile.cmake
+++ b/ports/openssl/unix/portfile.cmake
@@ -72,6 +72,10 @@ elseif(VCPKG_TARGET_IS_IOS)
     endif()
     # disable that makes linkage error (e.g. require stderr usage)
     list(APPEND CONFIGURE_OPTIONS no-ui no-asm)
+elseif(VCPKG_TARGET_IS_VISIONOS)
+    set(OPENSSL_ARCH darwin64-arm64)
+    # visionOS does not permit JIT or OpenSSL's platform assembly path.
+    list(APPEND CONFIGURE_OPTIONS no-ui no-asm)
 elseif(VCPKG_TARGET_IS_TVOS OR VCPKG_TARGET_IS_WATCHOS)
     set(OPENSSL_ARCH iphoneos-cross)
     # disable that makes linkage error (e.g. require stderr usage)
